daemon/restart.go

```diff
@@ -18,7 +18,7 @@ func (daemon *Daemon) ContainerRestart(name string, seconds *int) error {
 	if err != nil {
 		return err
 	}
-	if seconds == nil {
+	if seconds == nil || *seconds < 0 {
```

Comment:
Perhaps it would be better to check the validity of `-t` [here](https://github.com/moby/moby/blob/master/api/server/router/container/container_routes.go#L241-L249).